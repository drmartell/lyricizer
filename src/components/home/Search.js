import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ searchTerm, onChange, onClick }) => (
  <>
    <input type="text" placeholder="Search an artist" value={searchTerm} onChange={onChange} />
    <button onClick={onClick}>Search</button>
  </>
);

Search.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Search;
