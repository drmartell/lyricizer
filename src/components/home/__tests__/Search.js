import React from 'react';
import { shallow } from 'enzyme';
import Search from '../Search';

describe('Search Component', () => {
  it('should render Search', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
});
