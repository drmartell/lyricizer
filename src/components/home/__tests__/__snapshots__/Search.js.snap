// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render Search 1`] = `
<Fragment>
  <input
    onKeyDown={[Function]}
    placeholder="Search an artist"
    type="text"
  />
  <button>
    Search
  </button>
</Fragment>
`;
